transaction-view/src/bytes.rs

```diff
@@ -128,6 +128,35 @@ pub fn advance_offset_for_type<T: Sized>(bytes: &[u8], offset: &mut usize) -> Re
     Ok(())
 }
 
+/// Return a reference to the next slice of `T` in the buffer, checking bounds
```

Comment:
I think the unsafe code in this module needs more eyes. Isn't it possible you could have alignment issues with the pointers you are creating for these slices and structs?

Reply:
All types we're slicing are alignment 1, so it's fortunately not possible.

I think I dropped some of the const asserts I had originally for that. I will re-add some const asserts above each unsafe line for those conditions.

Reply:
bc9052981bad5109698599004b7eb01fcf43aa92 - made the reading functions unsafe since the constraints cannot be asserted on in the functions + asserts at call-sites.

Similarly (but separate PR) - #2685 adds call-site safety documentation for existing accessors.